cpp/locality/NeighborQuery.h

```diff
@@ -22,6 +22,23 @@
 
 namespace freud { namespace locality {
 
+//! Enumeration for types of queries.
+enum QueryType
```

Comment:
I moved this out of the scope of `QueryArgs`. I would prefer to use an `enum class` instead of an `enum` (since it is strongly typed) but support for that was only added to Cython somewhat recently, and I suspect it would cause us headaches in the interim. https://github.com/cython/cython/pull/3640

Reply:
Yeah I'm pretty sure I tried that (nested inside `QueryArgs`) when I first made this PR and found that limitation, didn't know they got around to adding the feature. We can change it later if we like, but I agree that it's not worth doing until Cython has proper support. There are other workarounds (using an additional namespace, nesting inside another class, etc) to enforce better scoping, but the benefits are pretty minimal.